lib/Site.js

```diff
@@ -133,13 +133,14 @@ Site.initSite = function (rootPath) {
   });
 };
 
-Site.prototype.readSiteConfig = function () {
+Site.prototype.readSiteConfig = function (baseUrl) {
   return new Promise((resolve, reject) => {
     const siteConfigPath = path.join(this.rootPath, SITE_CONFIG_NAME);
     fs.readJsonAsync(siteConfigPath)
       .then((config) => {
         this.siteConfig = config;
-        resolve(config);
+        this.siteConfig.baseUrl = baseUrl || this.siteConfig.baseUrl;
```

Comment:
I believe this won't work when `baseUrl` is an empty string.

Reply:
Ah yes my bad. see: https://stackoverflow.com/a/3088161